brili.ts

```diff
@@ -594,8 +594,12 @@ function evalInstr(instr: bril.Instruction, state: State): Action {
     let args = instr.args || [];
     let values = args.map(function (i) {
       let val = get(state.env, i);
-      if (Object.is(-0, val)) { return "-0.00000000000000000" };
-      if (typeof val == "number") { return val.toFixed(17) } else {return val.toString()}}
+      if (typeof val == "number") {
+        if ( Object.is(-0.0, val)) { return "-" + val.toFixed(17) }
+        else if (val != 0.0 && Math.abs(Math.log10(Math.abs(val))) >= 10) { return val.toExponential(6) }
```

Comment:
I guess this is the crux of where we need to make a decision: should this be `toExponential(17)` to match the precision of the non-scientific-notation case?